lib/linux_kernel/kthread.c

```diff
@@ -227,16 +227,15 @@ M0_INTERNAL int m0_thread_confine(struct m0_thread *q,
 
 		  put_task_struct(p);
 		*/
-
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
-		cpumask_copy(&p->cpus_mask, cpuset);
```

Comment:
is cpumask is not needed for kernel version above 4.18 for x8r6 arch ?

Reply:
This code is framed by andriy, it is used below

Reply:
@madhavemuri it is needed, and it is set below, on line 233 now. To make it clear, here is the new code:

```c
230 #if LINUX_VERSION_CODE < KERNEL_VERSION(4,18,0) || defined CONFIG_AARCH64
231                 cpumask_copy(&p->cpus_allowed, cpuset);
232 #else
233                 cpumask_copy(&p->cpus_mask, cpuset);
234 #endif
235 #if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
236                 p->rt.nr_cpus_allowed = nr_allowed;
237 #else
238                 p->nr_cpus_allowed = nr_allowed;
239 #endif
```

Reply:
ok.